### PR TITLE
feat: Add CLI option to set registries.conf.d path

### LINKED ISSUE #4593: Make `ctx.SystemRegistriesConfDirPath` configurable via CLI
**Description**

It is currently possible to use a specific `containers-registries.conf(5)` file via the hidden `--registries-conf` CLI option but there is no similar option to set the directory path used for `containers-registries.conf.d(5)`. Instead, the usual user path (if extant) or system path will be used.

**Steps to reproduce the issue:**
N/A

**Describe the results you received:**
N/A

**Describe the results you expected:**
N/A

**Additional information you deem important (e.g. issue happens only occasionally):**
None

**Output of `crio --version`:**

```
crio version 1.20.0
Version:       1.20.0
GitCommit:     d388528dbed26b93c5bc1c89623607a1e597aa57
GitTreeState:  clean
BuildDate:     2020-12-11T17:42:02Z
GoVersion:     go1.15.5
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by maybe-sybr**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This change adds a CLI option analogous to `--registries-conf` for setting the path to `registries.conf.d`.

#### Which issue(s) this PR fixes:
Closes #4593 

#### Special notes for your reviewer:
The name of the option was fairly obvious to choose but also happily mimics the choice buildah has made for the same option.

#### Does this PR introduce a user-facing change?
```release-note
A new CLI option `--registries-conf-dir` is added which allows users to select
a specific `registries.conf.d(5)` path which will be used instead of the user or
system default paths.
```

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM

Though I recommend to use the system defaults where possible.

**Comment by TomSweeneyRedHat** [comment]
Any man page changes/additions?

File: internal/criocli/criocli.go [hunk 1]
```diff
@@ -808,6 +808,14 @@ func getCrioFlags(defConf *libconfig.Config) []cli.Flag {
 			EnvVars:     []string{"CONTAINER_REGISTRIES_CONF"},
 			TakesFile:   true,
 		},
+		&cli.StringFlag{
+			Name:        "registries-conf-dir",
+			Usage:       "path to the registries.conf.d directory",
+			Destination: &defConf.SystemContext.SystemRegistriesConfDirPath,
+			Hidden:      true,
+			EnvVars:     []string{"CONTAINER_REGISTRIES_CONF_D"},
```
Comment (haircommander) [inline internal/criocli/criocli.go:816]: can this be `CONTAINERS_REGISTRIES_CONF_DIR` to be consistent with the other fields that take a directory?

**Comment by haircommander** [comment]
/ok-to-test

good point @TomSweeneyRedHat though I don't think they're needed because we're hiding the field

**Comment by haircommander** [comment]
thanks @maybe-sybr one nit but otherwise

/approve

**Comment by haircommander** [comment]
/retest

heya @chavafg, it seems we're running into issues with kata tests: jenkins.katacontainers.io/job/kata-containers-crio-PR/16565/console. Can you PTAL? Thanks as always

**Comment by haircommander** [comment]
test failures are not the fault of this PR btw

**Comment by chavafg** [comment]
hi @haircommander, 

I see the job has now passed :). Maybe it was just a random error.

**Comment by haircommander** [comment]
> hi @haircommander,
> 
> I see the job has now passed :). Maybe it was just a random error.

:upside_down_face: you're right, thank you for looking!

**Comment by maybe-sybr** [comment]
> Any man page changes/additions?

@TomSweeneyRedHat : AFAICT no, since this is a hidden argument. `--registries-conf` (the arg this new one is modelled on) isn't mentioned in the man page.

**Comment by maybe-sybr** [comment]
@haircommander - I've just made that minor change, rebased off the tip of master and re-pushed. Hopefully this will be good for merge now, sorry for the delay!

**Comment by haircommander** [comment]
no worries :) our CI is busted now anyway, so it'll be a bit longer. in the meantime
/approve
LGTM

**Pull request merged**